### ISSUE TITLE
Async methods

### ISSUE THREAD

**Issue by vector-man** [FrozenCow/sharpfilesystem#10]
It would be nice if the library supported async/await methods.

**Comment by bobvanderlinden** [comment]
Yes, I've thought about adding this as well. It would probably need a separate interface (`IFileSystemAsync`?) that always result in `Task<...>`. Existing `IFileSystem` implementations can be wrapped by a `IFileSystemAsync` implementation.
